pkg/utils/kubeutils/metadata.go

```diff
@@ -20,13 +20,16 @@ func FromKubeMeta(meta metav1.ObjectMeta) core.Metadata {
 	}
 }
 
-func ToKubeMeta(meta core.Metadata) metav1.ObjectMeta {
+func ToKubeMeta(meta *core.Metadata) metav1.ObjectMeta {
 	skMeta := ToKubeMetaMaintainNamespace(meta)
 	skMeta.Namespace = clients.DefaultNamespaceIfEmpty(meta.Namespace)
 	return skMeta
 }
 
-func ToKubeMetaMaintainNamespace(meta core.Metadata) metav1.ObjectMeta {
+func ToKubeMetaMaintainNamespace(meta *core.Metadata) metav1.ObjectMeta {
+	if meta == nil {
+		return metav1.ObjectMeta{}
+	}
```

Comment:
is this really preferable to returning nil?

Reply:
That could have happened before as well. The issue here is that the `metav1.ObjectMeta{}` is never a pointer in kube objects, so this is closer to how it will be used 99% of the time.